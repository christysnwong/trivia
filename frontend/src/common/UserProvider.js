import React, {useEffect, useState} from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  stats: {
    level: 14,
    title: "Pro",
    "quizzes completed": 34,
    points: 3440,
    remainingPts: 60,
    levelPts: 300,
  },
};

const UserProvider = ({
  children,
  currUser = demoUser
}) => {
  
  const [currentUser, setCurrUser] = useState();

  return (
    // <UserContext.Provider value={{ currUser }}>
    <UserContext.Provider value={{ currUser, setCurrUser }}>
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
