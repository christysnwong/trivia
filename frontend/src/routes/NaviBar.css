nav {
    background-color: white;
    box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.12), 0 1px 5px 0 rgba(0, 0, 0, 0.2);
}

.nav-item {
    /* text-align: center;
    margin-left: 50px; */
}

nav img {
    height: 35px;
    width: 35px;
    margin-right: 10px;
}

nav a {
    text-decoration: none;
}

nav a.active {
    font-weight: bold;
    color: black;
}


nav a:hover {
    text-decoration: none;
}

