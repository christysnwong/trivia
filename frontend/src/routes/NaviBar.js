import { useState, useContext } from "react";
import UserContext from "../common/UserContext";
import { NavLink as RRNavLink} from "react-router-dom";

import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from "reactstrap";

import "./NaviBar.css";

const NaviBar = ({ logout }) => {
  const { currUser } = useContext(UserContext);
  console.debug("Navibar - currUser", currUser);

  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);

  const userLinks = () => {
    return (
      <>
        <UncontrolledDropdown nav inNavbar>
          <DropdownToggle nav caret>
            My Dashboard
          </DropdownToggle>
          <DropdownMenu end>
            <DropdownItem header>
              <i className="fa-solid fa-star me-2"></i>
              {currUser.username} - Level {currUser.stats.level}{" "}
              {currUser.stats.title}
            </DropdownItem>
            <DropdownItem divider />
            <DropdownItem>
              <NavItem>
                <NavLink to="/profile" activeClassName="active" tag={RRNavLink}>
                  Profile
                </NavLink>
              </NavItem>
            </DropdownItem>
            <DropdownItem>
              <NavItem>
                <NavLink to="/stats" activeClassName="active" tag={RRNavLink}>
                  Statistics / Badges
                </NavLink>
              </NavItem>
            </DropdownItem>
            <DropdownItem>
              <NavItem>
                <NavLink
                  to="/personalbest"
                  activeClassName="active"
                  tag={RRNavLink}
                >
                  Personal Best Scores
                </NavLink>
              </NavItem>
            </DropdownItem>
            <DropdownItem>
              <NavItem>
                <NavLink
                  to="/sessions"
                  activeClassName="active"
                  tag={RRNavLink}
                >
                  Recent Played Sessions
                </NavLink>
              </NavItem>
            </DropdownItem>
          </DropdownMenu>
        </UncontrolledDropdown>
        <NavItem>
          <NavLink to="/favourites" activeClassName="active" tag={RRNavLink}>
            My Favourites
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink
            to="/logout"
            activeClassName="active"
            tag={RRNavLink}
            onClick={logout}
          >
            Log out
          </NavLink>
        </NavItem>
      </>
    );
  };

  const visitorLinks = () => {
    return (
      <>
        <NavItem>
          <NavLink to="/login" activeClassName="active" tag={RRNavLink}>
            Login
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink to="/signup" activeClassName="active" tag={RRNavLink}>
            Sign Up
          </NavLink>
        </NavItem>
      </>
    );
  };

  return (
    <Navbar expand="md" className="bg-sunset" container>
      <NavbarBrand href="/" className="brand">
        <img alt="logo" src="/brain.png" />
        Trivia Guru
      </NavbarBrand>

      <NavbarToggler onClick={toggle} />
      <Collapse isOpen={isOpen} navbar>
        <Nav className="me-auto" navbar>
          <NavItem>
            <NavLink to="/quizzes/" activeClassName="active" tag={RRNavLink}>
              Quizzes
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to="/leaderboard" activeClassName="active" tag={RRNavLink}>
              Leaderboard
            </NavLink>
          </NavItem>
        </Nav>
        <Nav navbar>{currUser == null ? visitorLinks() : userLinks()}</Nav>
      </Collapse>
      {/* </div> */}
    </Navbar>
  );
};

export default NaviBar;
