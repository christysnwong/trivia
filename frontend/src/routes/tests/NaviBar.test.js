import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NaviBar from "../NaviBar";
import UserProvider from "../../common/UserProvider";


it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <NaviBar />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <NaviBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currUser={null}>
        <NaviBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
