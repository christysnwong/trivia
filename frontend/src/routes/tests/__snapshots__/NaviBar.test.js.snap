// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-sunset navbar navbar-expand-md"
  >
    <div
      class="container"
    >
      <a
        class="brand navbar-brand"
        href="/"
      >
        <img
          alt="logo"
          src="/brain.png"
        />
        Trivia Guru
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="me-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/quizzes/"
            >
              Quizzes
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/leaderboard"
            >
              Leaderboard
            </a>
          </li>
        </ul>
        <ul
          class="navbar-nav"
        >
          <li
            class="dropdown nav-item"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              class="dropdown-toggle nav-link"
              href="#"
            >
              My Dashboard
            </a>
            <div
              aria-hidden="true"
              class="dropdown-menu dropdown-menu-end"
              role="menu"
              tabindex="-1"
            >
              <h6
                class="dropdown-header"
                tabindex="-1"
              >
                <i
                  class="fa-solid fa-star me-2"
                />
                testuser - Level 14 Pro
              </h6>
              <div
                class="dropdown-divider"
                tabindex="-1"
              />
              <button
                class="dropdown-item"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/profile"
                  >
                    Profile
                  </a>
                </li>
              </button>
              <button
                class="dropdown-item"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/stats"
                  >
                    Statistics / Badges
                  </a>
                </li>
              </button>
              <button
                class="dropdown-item"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/personalbest"
                  >
                    Personal Best Scores
                  </a>
                </li>
              </button>
              <button
                class="dropdown-item"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/sessions"
                  >
                    Recent Played Sessions
                  </a>
                </li>
              </button>
            </div>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/favourites"
            >
              My Favourites
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/logout"
            >
              Log out
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="bg-sunset navbar navbar-expand-md"
  >
    <div
      class="container"
    >
      <a
        class="brand navbar-brand"
        href="/"
      >
        <img
          alt="logo"
          src="/brain.png"
        />
        Trivia Guru
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="me-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/quizzes/"
            >
              Quizzes
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/leaderboard"
            >
              Leaderboard
            </a>
          </li>
        </ul>
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/login"
            >
              Login
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/signup"
            >
              Sign Up
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
